cpp/examples/tutorial_examples/dataset_example.cc

```diff
@@ -0,0 +1,141 @@
+#include <arrow/api.h>
+#include <arrow/csv/api.h>
+#include <parquet/arrow/reader.h>
+#include <parquet/arrow/writer.h>
+#include <arrow/io/api.h>
+#include <arrow/ipc/api.h>
+#include <arrow/pretty_print.h>
+#include <arrow/result.h>
+#include <arrow/status.h>
+#include <arrow/table.h>
+#include <arrow/compute/api.h>
+#include <arrow/filesystem/api.h>
+#include <arrow/dataset/api.h>
```

Comment:
Are all these header files necessary? Maybe can be simplified due to transitive inclusion.

Reply:
Removed all the headers I could without throwing errors.